pghoard/restore.py

```diff
@@ -396,9 +400,17 @@ def _get_basebackup(self, pgdata, basebackup, site,
             # Object is a raw (encrypted, compressed) basebackup
             basebackup_data_files = [[basebackup, -1]]
 
+        if tablespace_base_dir and os.path.exists(tablespace_base_dir):
+            # we just care that the dir exists, but we're OK if there are other objects there
+            raise RestoreError("Tablespace base directory does not {!r} exist, aborting."
+                               .format(tablespace_base_dir))
+
         # Map tablespaces as requested and make sure the directories exist
         for tsname, tsinfo in tablespaces.items():
             tspath = tablespace_mapping.pop(tsname, tsinfo["path"])
+            if tablespace_base_dir:
```

Comment:
This should be changed to `if tablespace_base_dir and not tspath:` to allow overriding paths per tablespace even if we're using a tablespace base dir option (or alternatively you should exit with an error if the user uses both options)